src/nix/hive/assets.rs

```diff
@@ -51,7 +51,10 @@ impl Assets {
             // We explicitly specify `path:` instead of letting Nix resolve
             // automatically, which would involve checking parent directories
             // for a git repository.
-            let uri = format!("path:{}", temp_dir.path().to_str().unwrap());
+            let uri = format!(
```

Comment:
I found a few usages of this: https://github.com/zhaofengli/colmena/pull/236

Reply:
Not sure if those are relevant, but my thinking was that it might resolve other future regressions.

Reply:
I did this near the point where the data (the temp path) leaves our control as this isn't a bug in the Rust side of things but at the boundary to a different system.
This only affects paths that we pass to nix (so ideally we would canonicalize the paths in [FlakeMetadata::resolve](https://github.com/zhaofengli/colmena/blob/main/src/nix/flake.rs#L88)). We could do that by passing something like a FlakePath to that function instead of the str.
```rs
enum FlakePath<'s> {
Path(&'s Path),
Git(&'s str)
}

<Display impl that canonicalizes before creating the `path:{}` string>
```
This would allow us to keep the simple temp dir paths in places where it could get debug printed without causing any irritation why the tmp dir looks weird. The symlinks for tempdirs on macos are an implementation detail and should not be exposed in level-1 debug output (such as a dbg!()) print, IMHO.